lib/toxiproxy.rb

```diff
@@ -211,6 +211,23 @@ def toxics
 
   private
 
+  def self.http_request(request)
+    ensure_webmock_whitelists_toxiproxy if defined? WebMock
+    http.request(request)
+  end
+
+  def http_request(request)
+    self.class.http_request(request)
+  end
+
+  def self.ensure_webmock_whitelists_toxiproxy
+    endpoint = "#{uri.host}:#{uri.port}"
+    WebMock::Config.instance.allow ||= []
```

Comment:
It took me a little bit to get comfortable with this appending every time instead of on boot, but it's the only thing I can think of that'll protect us against people emptying the whitelist at run-time in their test suite. Preventing that seems like an uphill battle.